applications/serial_lte_modem/src/slm_at_udp_proxy.c

```diff
@@ -355,7 +350,7 @@ static void udp_thread_func(void *p1, void *p2, void *p3)
 		if ((fds.revents & POLLIN) != POLLIN) {
 			continue;
 		} else {
-			ret = recvfrom(udp_sock, data, NET_IPV4_MTU, 0,
+			ret = recvfrom(udp_sock, rx_data, sizeof(rx_data), 0,
```

Comment:
This isn't a safe approach, we end up with two threads (TCP and UDP) writing to a single buffer. In case some data is received on both threads, we'll have concurrency issues.

If you want to save some bytes on buffers, perhaps you could repurpose the `data_hex` buffer, i. e. use it for `recvfrom()` call, but write only half of its size. Then if conversion to hex is needed you can still use it, just convert from the end to the start (so you don't overwrite the data with its hex representation).

Reply:
The prerequisite in SLM is that there is no concurrent socket RX, that's why we move to use global guaranteed buffer in .bss

Reply:
Not sure how do you want to guarantee that there's no concurrent RX on both threads, given we have the server here and the communication is client-initiated. But I'm not aware of are requirements, perhaps there are prerequisites for the clients as well.

Reply:
We did not include anything to prevent concurrent socket RX. :-)
Just as of now we try to save every byte for TLS server (maybe also DTLS server, soo)